app/src/main/java/org/wikipedia/history/HistoryViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package org.wikipedia.history
+
```

Comment:
Imho the ViewModel knows too much about the database implementation. We can move these Room-related methods to a new HistoryRepository class and inject it into the ViewModel constructor.

```suggestion

//new file
          interface HistoryRepository {
    suspend fun filterHistoryItems(searchQuery: String) : List<Any>
    suspend fun deleteHistory(entry: HistoryEntry)
    suspend fun deleteAllHistory()
    suspend fun insertHistory(entries: List<HistoryEntry>)
}
```

```suggestion
//new file
class HistoryRepositoryImpl(
    private val historyDao: HistoryEntryDao = AppDatabase.instance.historyEntryDao(),
    private val historyWithImageDao: HistoryEntryWithImageDao = AppDatabase.instance.historyEntryWithImageDao(),
    private val dispatcher: CoroutineDispatcher = Dispatchers.IO
): HistoryRepository {

    override suspend fun filterHistoryItems(searchQuery: String) = withContext(dispatcher) {
        historyWithImageDao.filterHistoryItems(searchQuery)
    }
    override suspend fun deleteHistory(entry: HistoryEntry) = historyDao.delete(entry)
    override suspend fun deleteAllHistory() = historyDao.deleteAll()
    override suspend fun insertHistory(entries: List<HistoryEntry>) = historyDao.insert(entries)
}
```
By doing this, the ViewModel becomes more testable, and we have a centralized place to handle these history-database operations. This improves the unit testability of the ViewModel. Since DAOs are interfaces, we can also create fakes to unit test the repository.